x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearchUsageTransportAction.java

```diff
@@ -175,7 +176,9 @@ public void onResponse(IndicesStatsResponse indicesStatsResponse) {
                     Map<String, IndexStats> indicesStats = indicesStatsResponse.getIndices();
                     int queryRulesetCount = indicesStats.values()
                         .stream()
-                        .mapToInt(indexShardStats -> (int) indexShardStats.getPrimaries().getDocs().getCount())
+                        .map(indexShardStats -> indexShardStats.getPrimaries().getDocs())
+                        .filter(Objects::nonNull)
+                        .mapToInt(docsStats -> (int) docsStats.getCount())
```

Comment:
I observed NPE from this line in serverless QA. The similar NPE has also been [observed](https://github.com/elastic/elasticsearch/issues/108890) and [fixed](https://github.com/elastic/elasticsearch/pull/108891) for another callsite. Hence I am fixing a few more places that could have the same potential problem.

All fields in `org.elasticsearch.action.admin.indices.stats.CommonStats` are nullable which feels a bit trappy and verbose since null check needs to be done in each downstream call. The check is indeed performed in most places. But I wonder whether having more meaningful Empty values for these sub-stats could hav been better. Certainly out of scope for this PR.